### PR TITLE
Redo ban system / Asset validation

### PR DESCRIPTION
**Pull request by Z1Meme**

### CODE REVIEW

**Comment by Z1Meme** [comment]
@QuentinGruber I fully tested this but I need a second opinion when you have time. Old zoneservers will still allow connections with the updated loginserver, but new bans won't work, only ones already registered. The main thing is that the loginserver no longer handles any bans beside global, it's handled on the zone's side entirely, and the loginserver gets that information from the zone before allowing a connection.

**Comment by QuentinGruber** [comment]
> @QuentinGruber I fully tested this but I need a second opinion when you have time. Old zoneservers will still allow connections with the updated loginserver, but new bans won't work, only ones already registered. The main thing is that the loginserver no longer handles any bans beside global, it's handled on the zone's side entirely, and the loginserver gets that information from the zone before allowing a connection.

checking that tomorow :)

**Comment by QuentinGruber** [comment]
We will probably need to modify stuff in live-login-server https://github.com/H1emu/live-login-server

File: src/packets/LoginUdp/LoginUdp_11/loginpackets.ts [hunk 1]
```diff
@@ -377,6 +377,28 @@ const packets: PacketStructures = [
         { name: "message", type: "string", defaultValue: "" }
       ]
     }
+  ],
+  [
+    "H1emu.HadesInit",
```
Comment (QuentinGruber) [inline src/packets/LoginUdp/LoginUdp_11/loginpackets.ts:382]: Hades ?

**Comment by QuentinGruber** [comment]
> @QuentinGruber I fully tested this but I need a second opinion when you have time. Old zoneservers will still allow connections with the updated loginserver, but new bans won't work, only ones already registered. The main thing is that the loginserver no longer handles any bans beside global, it's handled on the zone's side entirely, and the loginserver gets that information from the zone before allowing a connection.

For me it's good and i don't care that this update isn't backward compatible with old zoneservers. The fact that the zoneserver is the one who decide is probably a better idea, but how this will work for ip bans ? Or hwid bans that were on the login server side ? We bundle that all in "global bans" ?

**Comment by Z1Meme** [comment]
> > @QuentinGruber I fully tested this but I need a second opinion when you have time. Old zoneservers will still allow connections with the updated loginserver, but new bans won't work, only ones already registered. The main thing is that the loginserver no longer handles any bans beside global, it's handled on the zone's side entirely, and the loginserver gets that information from the zone before allowing a connection.
> 
> For me it's good and i don't care that this update isn't backward compatible with old zoneservers. The fact that the zoneserver is the one who decide is probably a better idea, but how this will work for ip bans ? Or hwid bans that were on the login server side ? We bundle that all in "global bans" ?

HWID and IP bans are already handled separately from global, the loginserver tells the zone if a client is ip or hwid banned or using a vpn, and the zone tells the login if it wants to accept that client or not.

File: src/packets/LoginUdp/LoginUdp_11/loginpackets.ts (hunk 1 above)
Comment (Z1Meme) [reply src/packets/LoginUdp/LoginUdp_11/loginpackets.ts:382]: That's what Jamie named the asset validation stuff, HadesInit isn't really used rn so we just use HadesQuery to start the validation as soon as a client connects to the login.

**Comment by Z1Meme** [comment]
> We will probably need to modify stuff in live-login-server https://github.com/H1emu/live-login-server

This should be fine, isClientVerified / HwidBanned didn't change and still serve the same purpose.

**Comment by avcio9** [comment]
@QuentinGruber Looking for contact, can you check discord?

**Comment by QuentinGruber** [comment]
@avcio9 yep i will be back on discord in like 2-3h ( i don't have it on my phone )

**Pull request merged**